src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/permissionSelection.jsx

```diff
@@ -0,0 +1,181 @@
+ *
+ *
+ * Updating the Form Field Value
+ *
+ *    In addition to updating the state, when a value is changed this component
+ *    recalculates the full list of API Scopes that need to be passed to the API.
+ *
+ *    So if the User has changed Project to "Admin" and Organization to "Read & Write",
+ *    we end up with a `state.permissions` like:
+ *
+ *       {
+ *         'Project': 'admin',
+ *         'Organization': 'write',
+ *       }
+ *
+ *    From there, we calculate the full list of API Scopes. This list includes all
+ *    implied scopes, meaning the above state would result in:
+ *
+ *       ['project:read', 'project:write', 'project:admin', 'org:read', 'org:write']
+ *
+ */
+export default class PermissionSelection extends AsyncView {
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+    form: PropTypes.object,
+  };
+
+  static propTypes = {
+    scopes: PropTypes.array.isRequired
+  };
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      permissions: {
+        Project: 'no-access',
+        Team: 'no-access',
+        Release: 'no-access',
+        Event: 'no-access',
+        Organization: 'no-access',
+        Member: 'no-access',
+      },
+    };
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.state.permissions = this.scopeListToPermissionState();
+  }
+
+  /**
+   * Converts the list of raw API scopes passed in to an object that can
+   * before stored and used via `state`. This object is structured by
+   * resource and holds "Permission" values. For example:
+   *
+   *    {
+   *      'Project': 'read',
+   *      ...,
+   *    }
+   *
+   */
+  scopeListToPermissionState() {
+    return new ConsolidatedScopes(this.props.scopes).toResourcePermissions();
+  }
+
+  /**
+   * Converts the "Permission" values held in `state` to a list of raw
+   * API scopes we can send to the server. For example:
+   *
+   *    ['org:read', 'org:write', ...]
+   *
+   */
+  permissionStateToList() {
+    const {permissions} = this.state;
+    const findResource = r => find(SENTRY_APP_PERMISSIONS, ['resource', r]);
+
+    return flatMap(
+      Object.entries(permissions),
+      ([r, p]) => findResource(r).choices[p].scopes
+    );
+  }
+
+  onChange = (resource, choice) => {
+    const {permissions} = this.state;
+
+    permissions[resource] = choice;
+
+    this.setState({permissions});
+    this.context.form.setValue('scopes', this.permissionStateToList());
+  };
+
+  renderBody() {
+    const {permissions} = this.state;
+
+    return (
+      <React.Fragment>
+        {SENTRY_APP_PERMISSIONS.map(config => {
+          const toChoice = ([value, opt]) => [value, opt.label];
+          const choices = Object.entries(config.choices).map(toChoice);
+          const value = permissions[config.resource];
+
+          return (
+            <SelectField
+              // These are not real fields we want submitted, so we use
```

Comment:
not ideal that we have to do it this way, but the comments really helped so tyty for that 😸

Reply:
Yea 😒 I tried doing this directly with `SelectControl` and some label stuff, but I had no idea how all of those things worked with Flexbox so it looked jacked up. I wasted enough time trying to make it work that I just went back to this.